elements/lisk-tree/src/sparse_merkle_tree/sparse_merkle_tree.ts

```diff
@@ -35,8 +43,108 @@ export class SparseMerkleTree {
 	public get db(): Database {
 		return this._db;
 	}
+
+	public async getNode(nodeHash: Buffer): Promise<TreeNode> {
+		const data = await this._hashToValueMap.get(nodeHash);
+
+		if (!data) {
+			throw new Error(`Hash does not exist in merkle tree: ${nodeHash.toString('hex')}`);
+		}
+
+		if (isLeaf(data)) {
+			const { key, value } = getLeafData(data, this.keyLength);
+			return new Leaf(key, value);
+		}
+
+		const { leftHash, rightHash } = getBranchData(data);
+
+		return new Branch(leftHash, rightHash);
+	}
+
+	public async update(key: Buffer, value: Buffer) {
+		if (!value) {
+			throw new Error('Value cannot be empty');
+		}
+
+		if (key.length !== this.keyLength) {
```

Comment:
Would prefer to use `byteLength` instead of `length` consistently when `Buffer` type is involved. It gave more readability and consistent with NodeJS docs as well.